### ISSUE TITLE
How to handle Samba changes

### ISSUE THREAD

**Issue by DLLarson** [davestephens/ansible-nas#224]
Hi,

First off, thanks for this great project! I've been wanting to ditch my FreeNAS setup for Linux server for quite some time. This project looks like just-the-ticket!

I would like to use a different layout for Samba than the default configuration. I really don't like exposing all those shares without security to the LAN. Also, many of the shares don't apply to my situation.

My impression from the docs is that I should make changes to my own `ansible-nas/inventory/my-ansible-nas/group_vars/nas.yml` file. This seems to work fine for additive or overlaying changes. But what if I want to remove some of the configurations that are defined in `ansible-nas/group_vars/all.yml`?

I'm definitely not an Ansible expert and am willing to learn but I don't see a way to essentially disable the samba definitions in `all.yml`.

Can anyone offer any insights? Is this a case where I need to edit `all.yml`?

-Dale

**Comment by davestephens** [comment]
Glad you're liking the look of the project!

You'd simply override `samba_shares` in your `nas.yml` - whatever is in
there will take prescedence over `all.yml`.

The the config in `all.yml` is the base config, any variables defined in
`nas.yml` "win", no combining takes place of individual variables / lists
etc.

We actually use an Ansible role called `bertvv.samba` for all the Samba
stuff - if you look that up you'll see there's a whole bunch more config
you can take advantage of simply by specifying the relevant variables in
your `nas.yml`.

**Comment by DLLarson** [comment]
I must be a bit thick up stairs but I don't see how I can use `nas.yml` make the following `all.yml` not happen at all. Lets say I don't want the following `downloads` and `movies` `all.yml` definitions created by the `bertvv.samba` role:

```yaml
# Shares you want published over Samba.
samba_shares:
  - name: downloads
    comment: 'Stuff downloaded'
    guest_ok: yes
    public: yes
    writable: yes
    browsable: yes
    path: "{{ downloads_root }}"

  - name: movies
    comment: 'Movies'
    guest_ok: yes
    public: yes
    writable: yes
    browsable: yes
    path: "{{ movies_root }}"
# : 
# : 
```

I do see how I can define new ones or modify existing values. 

I just can't see how to simply remove undesirable definitions I don't want without deleting them from `all.yml`. 

For instance, would I have to redefine the unwanted ones completely again with empty text?

-Dale

**Comment by davestephens** [comment]
You simply set `samba_shares` in `nas.yml` to the shares you want created.

The resulting set of shares is not `samba_shares` from `all.yml` + `nas.yml`, it's either/or. By Ansible's order of prescedence, `nas.yml` wins - so whatever you put in there is what gets created.

Say you only wanted the tv and music shares, and wanted to throw away the rest, in `nas.yml` you set `samba_shares` to:

```
samba_shares:
  - name: tv
    comment: 'TV Episodes'
    guest_ok: yes
    public: yes
    writable: yes
    browsable: yes
    path: "{{ tv_root }}"

  - name: music
    comment: 'Music'
    guest_ok: yes
    public: yes
    writable: yes
    browsable: yes
    path: "{{ music_root }}"
```
...and that's it.

**Comment by DLLarson** [comment]
Ahhhh! 

So defining at a particular level of definition causes Ansible to ignore all those at that level a below.

That *is* simple. I gotta learn more about Ansible. It looks pretty sweet.

Thanks for the help!

I've been testing all this out in a VMWare VM before I commit to changing my FreeNAS box over to it. Fortunately it looks like I can boot either OS via USB drives and gain access to my ZFS raid array from either. This is exciting stuff!

Docker is much better than BSD jails. And it turns out there are Docker images for most of what I used on FreeNAS beyond the large list that you provide via Ansible.

On FreeNAS I had to run VirtualBox with a docker system inside of a linux VM. A total pain.

The only reason I used FreeNAS was to have ZFS. That's no longer an issue in Linux.

Thanks again!
-Dale
